java/microdonuts/src/main/java/com/otsample/api/ApiContextHandler.java

```diff
@@ -1,153 +1,147 @@
 package com.otsample.api;
 
+import com.google.gson.Gson;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonObject;
+import com.otsample.api.resources.DonutRequest;
+import com.otsample.api.resources.StatusReq;
+import com.otsample.api.resources.StatusRes;
+import io.opentracing.Scope;
 import io.opentracing.Span;
+import io.opentracing.contrib.web.servlet.filter.TracingFilter;
+import io.opentracing.util.GlobalTracer;
 import java.io.IOException;
 import java.io.PrintWriter;
+import java.util.EnumSet;
 import java.util.Properties;
 import java.util.UUID;
-
+import javax.servlet.DispatcherType;
 import javax.servlet.ServletException;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
-
+import org.eclipse.jetty.servlet.FilterHolder;
 import org.eclipse.jetty.servlet.ServletContextHandler;
 import org.eclipse.jetty.servlet.ServletHolder;
 
-import com.google.gson.Gson;
-import com.google.gson.JsonArray;
-import com.google.gson.JsonObject;
-import com.otsample.api.resources.DonutRequest;
-import com.otsample.api.resources.StatusReq;
-import com.otsample.api.resources.StatusRes;
+public class ApiContextHandler extends ServletContextHandler {
+  Properties config;
+  KitchenConsumer kitchenConsumer;
 
-import io.opentracing.Scope;
-import io.opentracing.util.GlobalTracer;
+  public ApiContextHandler() {
+
+    TracingFilter tracingFilter = new TracingFilter(GlobalTracer.get());
+    addFilter(new FilterHolder(tracingFilter), "/*", EnumSet.allOf(DispatcherType.class));
```

Comment:
adding this filter fixed issue.

Reply:
can you describe a little more about what this filter is and why it's needed?

Reply:
this filter extract span context from request headers and creates a new span using extracted context as parent.
created span is active until request is finished. 
without this filter client span and server spans are not connected.